packages/dev/core/src/Misc/textureTools.ts

```diff
@@ -253,15 +252,13 @@ const ProcessAsync = async (
         lodPostProcess = new PostProcess("lodCube", "lodCube", ["lod", "gamma"], null, 1.0, null, Texture.NEAREST_NEAREST_MIPNEAREST, engine, false, faceDefines[face]);
     }
 
-    if (!lodPostProcess.getEffect().isReady()) {
-        // Try again later
-        lodPostProcess.dispose();
-
-        setTimeout(() => {
-            ProcessAsync(texture, width, height, face, channels, lod, resolve, reject);
-        }, 250);
-
-        return;
+    let isPostProcessReady = lodPostProcess.getEffect().isReady();

```

Comment:
You can use `postProcess.getEffect().executeWhenCompiled` instead of the `while`.

See how it is done in `ApplyPostProcess` in the same file:

https://github.com/BabylonJS/Babylon.js/blob/master/packages/dev/core/src/Misc/textureTools.ts#L136-L160